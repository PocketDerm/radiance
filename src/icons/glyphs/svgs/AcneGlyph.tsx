import * as React from 'react';

function SvgAcneGlyph(props: React.SVGProps<SVGSVGElement>) {
  return (
    <svg
      viewBox="0 0 48 48"
      fill="none"
      xmlns="http://www.w3.org/2000/svg"
      {...props}
    >
      <path
        fillRule="evenodd"
        clipRule="evenodd"
        d="M21.993 10.023c0 1.123.896 2.033 2.012 2.033s2.02-.92 2.02-2.033A2.03 2.03 0 0024.006 8c-1.107 0-2.012.9-2.012 2.023zm1.203 0a.81.81 0 01.809-.813c.452 0 .818.358.818.813a.826.826 0 01-.818.823.818.818 0 01-.809-.823zm-1.203 27.944c0 1.123.896 2.033 2.012 2.033s2.02-.92 2.02-2.033a2.03 2.03 0 00-2.02-2.023c-1.107 0-2.012.9-2.012 2.023zm1.203 0a.81.81 0 01.809-.813c.452 0 .818.368.818.813a.826.826 0 01-.818.823.818.818 0 01-.809-.823zm14.793-12.04a2.027 2.027 0 01-2.021-2.033 2.016 2.016 0 114.032 0c0 1.113-.895 2.032-2.011 2.032zm0-2.846a.818.818 0 100 1.635.812.812 0 00.808-.822.803.803 0 00-.808-.813zM8 23.893c0 1.122.905 2.032 2.021 2.032 1.116 0 2.021-.92 2.021-2.032a2.025 2.025 0 00-2.02-2.023c-1.117 0-2.022.9-2.022 2.023zm1.203 0a.818.818 0 111.635.009.818.818 0 01-1.635-.01z"
        fill="currentColor"
      />
    </svg>
  );
}

export default SvgAcneGlyph;
