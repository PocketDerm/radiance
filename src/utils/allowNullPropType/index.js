export default wrapper => (props, propName, ...rest) =>
  props[propName] === null ? null : wrapper(props, propName, ...rest);