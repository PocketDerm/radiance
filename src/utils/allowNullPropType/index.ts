export default (wrapper: any) => (
  props: any,
  propName: string,
  ...rest: any[]
) => (props[propName] === null ? null : wrapper(props, propName, ...rest));
