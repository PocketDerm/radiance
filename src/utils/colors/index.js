import tinycolor from 'tinycolor2';

function percentToInt(str) {
  // '20%' -> 20
  return parseInt(str.replace(/%/g, ''), 10);
}

export function lighten(color, modifier) {
  return tinycolor(color)
    .lighten(percentToInt(modifier))
    .toString();
}

export function darken(color, modifier) {
  return tinycolor(color)
    .darken(percentToInt(modifier))
    .toString();
}

export function transparentize(color, modifier) {
  return tinycolor(color)
    .setAlpha(modifier)
    .toString();
}
