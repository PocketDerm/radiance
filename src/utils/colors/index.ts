import tinycolor from 'tinycolor2';

function percentToInt(str: string) {
  // '20%' -> 20
  return parseInt(str.replace(/%/g, ''), 10);
}

type ColorManipulationFunctionType = (
  color: string,
  modifier: string
) => string;

export const lighten: ColorManipulationFunctionType = (color, modifier) =>
  tinycolor(color)
    .lighten(percentToInt(modifier))
    .toString();

export const darken: ColorManipulationFunctionType = (color, modifier) =>
  tinycolor(color)
    .darken(percentToInt(modifier))
    .toString();

export const transparentize = (color: string, modifier: number) =>
  tinycolor(color)
    .setAlpha(modifier)
    .toString();
