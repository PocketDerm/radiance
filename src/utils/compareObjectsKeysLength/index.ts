const compareObjectsKeysLength = (
  object1: Record<string, unknown>,
  object2: Record<string, unknown>,
) => Object.keys(object1).length === Object.keys(object2).length;

export default compareObjectsKeysLength;
