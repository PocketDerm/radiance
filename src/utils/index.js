export * from './colors';
export { default as throwOnUndefinedProperty } from './throwOnUndefinedProperty';
