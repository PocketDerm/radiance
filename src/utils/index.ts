export * from './colors';
export { default as throwOnUndefinedProperty } from './throwOnUndefinedProperty';
export { default as withDeprecationWarning } from './withDeprecationWarning';
export { default as isObject } from './isObject';
export { applyPrimaryThemeVerticalOffset } from './themeStyles';
export * from './injectGlobalStyles/style';
