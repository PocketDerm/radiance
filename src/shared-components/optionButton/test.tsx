import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import { decorateWithThemeProvider } from 'tests/utils/decorateWithThemeProvider';

import AcneGlyph from '../../svgs/glyphs/acne-glyph.svg';

import { OptionButton } from './index';

describe('<OptionButton />', () => {
  const DecoratedOptionButton = decorateWithThemeProvider(OptionButton);

  describe('UI snapshots', () => {
    it('checkbox selected, without custom icon', () => {
      const tree = renderer
        .create(
          <DecoratedOptionButton
            selected
            onClick={() => undefined}
            text="checkbox text"
            optionType="checkbox"
            buttonType="primary"
          />,
        )
        .toJSON();

      expect(tree).toMatchSnapshot();
    });

    it('radio unselected, with icon node prop', () => {
      const tree = renderer
        .create(
          <DecoratedOptionButton
            onClick={() => undefined}
            text="radio text"
            optionType="radio"
            icon={<AcneGlyph />}
            buttonType="secondary"
          />,
        )
        .toJSON();

      expect(tree).toMatchSnapshot();
    });
  });

  describe('onClick callback', () => {
    it('is invoked when clicked', () => {
      const spy = jest.fn();
      const wrapper = mount(
        <DecoratedOptionButton
          onClick={spy}
          text="checkbox text"
          optionType="checkbox"
        />,
      );

      wrapper.simulate('click');
      expect(spy).toHaveBeenCalled();
    });
  });
});
