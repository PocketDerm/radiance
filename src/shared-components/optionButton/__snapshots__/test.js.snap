// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OptionButton /> UI snapshots checkbox selected, without custom icon 1`] = `
.emotion-8 {
  background-color: #ffffff;
  border: 1px solid #ededf0;
  box-shadow: 0 1px 3px 0 rgba(45,45,48,0.06);
  cursor: pointer;
  -webkit-transition: box-shadow 200ms;
  transition: box-shadow 200ms;
}

.emotion-8:hover {
  box-shadow: 0 1px 3px 0 rgba(45,45,48,0.15);
}

.emotion-6 {
  padding: 1.5rem;
}

@media (min-width:992px) {
  .emotion-6 {
    padding-left: 2.5rem;
    padding-right: 2.5rem;
  }
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row nowrap;
  -ms-flex-flow: row nowrap;
  flex-flow: row nowrap;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-0 {
  border: 2px solid;
  border-color: #332e54;
  background: #ffffff;
  width: 32px;
  height: 32px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row nowrap;
  -ms-flex-flow: row nowrap;
  flex-flow: row nowrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-transition: all 350ms;
  transition: all 350ms;
  border-radius: 4px;
  background: #332e54;
}

.emotion-0 svg {
  opacity: 0;
  -webkit-transition: all 350ms;
  transition: all 350ms;
}

.emotion-0 svg {
  opacity: 1;
  color: #ffffff;
  fill: #ffffff;
}

.emotion-2 {
  color: #332e54;
  margin-left: 1rem;
}

<div
  className="emotion-8 emotion-9"
  onClick={[Function]}
  type="clickable"
>
  <div
    className="emotion-6 emotion-7"
  >
    <div
      className="emotion-4 emotion-5"
    >
      <div
        className="emotion-0 emotion-1"
        selected={true}
      >
        <svg
          height={16}
          width={16}
        />
      </div>
      <p
        className="emotion-2 emotion-3"
      >
        checkbox text
      </p>
    </div>
  </div>
</div>
`;

exports[`<OptionButton /> UI snapshots radio unselected, with icon node prop 1`] = `
.emotion-9 {
  background-color: #ffffff;
  border: 1px solid #ededf0;
  box-shadow: 0 1px 3px 0 rgba(45,45,48,0.06);
  cursor: pointer;
  -webkit-transition: box-shadow 200ms;
  transition: box-shadow 200ms;
}

.emotion-9:hover {
  box-shadow: 0 1px 3px 0 rgba(45,45,48,0.15);
}

.emotion-7 {
  padding: 1.5rem;
}

@media (min-width:992px) {
  .emotion-7 {
    padding-left: 2.5rem;
    padding-right: 2.5rem;
  }
}

.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row nowrap;
  -ms-flex-flow: row nowrap;
  flex-flow: row nowrap;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-1 {
  border: 2px solid;
  border-color: #a6a1e2;
  background: #ffffff;
  width: 32px;
  height: 32px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row nowrap;
  -ms-flex-flow: row nowrap;
  flex-flow: row nowrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-transition: all 350ms;
  transition: all 350ms;
  border-radius: 50%;
  width: 48px;
  height: 48px;
}

.emotion-1 svg {
  opacity: 0;
  -webkit-transition: all 350ms;
  transition: all 350ms;
}

.emotion-1 svg {
  opacity: 1;
  color: #a6a1e2;
  fill: #a6a1e2;
}

.emotion-0 {
  display: block;
  -webkit-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
  -webkit-transition: color 350ms,-webkit-transform 350ms;
  -webkit-transition: color 350ms,transform 350ms;
  transition: color 350ms,transform 350ms;
}

.emotion-3 {
  color: #a6a1e2;
  margin-left: 1rem;
}

<div
  className="emotion-9 emotion-10"
  onClick={[Function]}
  type="clickable"
>
  <div
    className="emotion-7 emotion-8"
  >
    <div
      className="emotion-5 emotion-6"
    >
      <div
        className="emotion-1 emotion-2"
        selected={false}
      >
        <svg
          className="emotion-0"
          height={16}
          rotate={0}
          viewBox="0 0 45 45"
          width={16}
        >
          <path
            d="M23 33.213a1.213 1.213 0 1 1 0-2.427 1.213 1.213 0 0 1 0 2.427M23 29a3 3 0 1 0 0 6 3 3 0 0 0 0-6m0-14.788a1.213 1.213 0 1 1 0-2.427 1.213 1.213 0 0 1 0 2.427M23 10a3 3 0 1 0 0 6 3 3 0 0 0 0-6m10.213 13a1.213 1.213 0 1 1-2.427 0 1.213 1.213 0 0 1 2.427 0M29 23A3 3 0 1 0 35 23 3 3 0 0 0 29 23m-14.787 0a1.214 1.214 0 1 1-2.427-.001 1.214 1.214 0 0 1 2.427 0M10 23A3 3 0 1 0 16 23a3 3 0 0 0-6 0"
            fill="currentColor"
            fillRule="evenodd"
          />
        </svg>
      </div>
      <p
        className="emotion-3 emotion-4"
      >
        radio text
      </p>
    </div>
  </div>
</div>
`;
